### DISCUSSION TITLE
Disabling chroot

### DISCUSSION THREAD

**Discussion by sagarkadam04** [drakkan/sftpgo#938] [Q&A] [answered]
Can we disable chroot in sftpgo instead of using prefix ?

Eg:  If i create user named ftpusr with it's folder path /app/data/ftpusr and another user named ftpusr2 with /app/data/ftpusr2.

While accessing these folders via script, I have to provide path till  /app/data/ for both the users, So there is no use of prefix here.
I want to avoid this chroot issue, while migrate many existing users which are not chrooted to sftpgo.

Thank you drakkan for such wonderful utility, Please let me know if i made any mistake understanding this prefix logic.

**Comment by drakkan** [comment]
Hi,

chroot cannot be disabled but I'm not sure to understand your question. Are you using local filesystem or a cloud backend?

If you set the home dir for `ftpusr` to `/app/data/ftpusr` and you have the file `/app/data/ftpusr/file.txt` you can download it using `/file.txt` as path, you don't need to add the `/app/data/ftpusr` prefix. Paths are relative to the defined home dir for SFTP/FTP clients

**Comment by sagarkadam04** [comment]
Thanks for replying.

We are using SFTP server which is not chrooted, We use it to fetch/push data in various automation activity.
We are using local filesystem.

Because the SFTP server is not chrooted we implement previously, there are more than 100 scripts in which full path is mentioned like for `ftpusr` they have used `get /app/data/ftpusr/file.txt`

When i use SFTPGo and set prefix to `/app/data/`
Their command in script will be like `get /app/data/file.txt`  instead of `/app/data/ftpusr/file.txt`

So If we want to migrate to SFTPGO from our SFTP server, we have to make changes to 100+ scripts, so prefix is not useful here.
Please let me know what is the use of prefix ?

**Reply by drakkan** [reply]
> Thanks for replying.
> 
> We are using SFTP server which is not chrooted, We use it to fetch/push data in various automation activity. We are using local filesystem.
> 
> Because the SFTP server is not chrooted we implement previously, there are more than 100 scripts in which full path is mentioned like for `ftpusr` they have used `get /app/data/ftpusr/file.txt`
> 
> When i use SFTPGo and set prefix to `/app/data/` Their command in script will be like `get /app/data/file.txt` instead of `/app/data/ftpusr/file.txt`
> 
> So If we want to migrate to SFTPGO from our SFTP server, we have to make changes to 100+ scripts, so prefix is not useful here. Please let me know what is the use of prefix ?

just set the home dir to "/" and the start directory to `/app/data/ftpusr`, `/app/data/ftpusr1` and so on. 

Another, more secure, way is to set a different home directory for each user and the start directory as explained above, for this to work as you expect you have to move your files to the new locations

**Comment by sagarkadam04** [comment, marked as answer]
I was using SFTPGo 2.2.2 version so haven't seen start directory feature. I tried both the methods suggested by you and both are working great especially the start directory, it resolved my query.

![sftp](https://user-images.githubusercontent.com/110017874/181735879-28d87dce-13e6-44b6-8de8-99cce680ba72.jpg)
![start_and_home_dir](https://user-images.githubusercontent.com/110017874/181735908-93d703e4-9ef7-4694-88cc-6997a73ad55e.JPG)

Thank you for constantly updating this project.

**Reply by drakkan** [reply]
Glad you resolved your issue. If you can, please consider supporting the project with a [sponsorship/donation](https://github.com/drakkan/sftpgo#sponsors). 

I cannot continue to dedicate all my free time to SFTPGo forever without receiving anything back (at least I received a "thank you" from you, in general it's not so obvious)
